### PR TITLE
Test suite

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

File: Source/ARTRest.m [hunk 1]
```diff
@@ -530,6 +530,9 @@ BOOL ARTstartHandlingUncaughtExceptions(ARTRest *self) {
 }
 
 void ARTstopHandlingUncaughtExceptions(ARTRest *self) {
+    if (!self) {
```
Comment (tcard) [inline Source/ARTRest.m]: `self` should never be null here. When is it?

**Comment by tcard** [comment]
I'm merging this since it'll improve stability, but I'd like to understand why this happens at some point.

File: Source/ARTRest.m (hunk 1 above)
Comment (ricardopereira) [inline Source/ARTRest.m]: @tcard Sorry, I can't recall when it was null but it wasn't in a normal state.

Comment (ricardopereira) [inline Source/ARTRest.m]: @tcard Another thing. I was facing a deadlock as well. I spent some time at my lunch time to figure it out but the time wasn't enough. I will open a new PR.

**Pull request merged**